### PR TITLE
Responsive design and UI changes

### PR DESCRIPTION
**Pull request by Denis-Dinkov**

### CODE REVIEW

File: src/assets/images/icons/icon/pc.svg
Comment (pivanov) [inline src/assets/images/icons/icon/pc.svg:1]: can we add it with 24x24 size?

File: src/components/burgerMenu/index.tsx [hunk 1]
```diff
@@ -25,66 +33,71 @@ const BurgerMenu = () => {
     setIsOpen(!isOpen);
   }, [isOpen]);
 
+  useEventBus<IEventBusClickLink>('@@-click-link', () => {
```
Comment (pivanov) [inline src/components/burgerMenu/index.tsx:36]: `@@-click-link` is to generic

File: src/components/mobileChainSelect/index.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  return Object.values(SUPPORTED_CHAIN_GROUPS)
+    .map((group) => ({
+      ...group,
+      chains: group.chains.filter((chain) =>
+        chain.name.toLowerCase().includes(query.toLowerCase()),
+      ),
+    }))
+    // Only include groups that have chains matching the query
+    .filter((group) => group.chains.length > 0);
+};
```
Comment (pivanov) [inline src/components/mobileChainSelect/index.tsx:36]: map => filter ... we can use for loop or reduce

File: src/components/mobileChainSelect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+              onClick={handleMenuClick}
+            >
+              <Icon name="icon-close" />
+            </button>
+          </div>
+
+          <SearchChain />
+
+          {
+            Array.from(filteredGroups) && filteredGroups.length > 0
```
Comment (pivanov) [inline src/components/mobileChainSelect/index.tsx:145]: that expression is redundant and unnecessary

File: src/components/navigation/navigationDropdown.tsx [hunk 4]
```diff
@@ -11,29 +17,54 @@ import {
...
+
+  const toggleDropdown = useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen]);
+
+  const handleClickOutside = useCallback((e: MouseEvent) => {
+    if (dropdownRef.current && !dropdownRef.current.contains(e.target as Node)) {
+      setIsOpen(false);
+    }
+  }, []);
```
Comment (pivanov) [inline src/components/navigation/navigationDropdown.tsx:45]: check already created hook `useOnClickOutside`

File: src/components/navigation/navigationItem.tsx [hunk 5]
```diff
@@ -19,7 +24,22 @@ export const NavigationItem = ({
   linkProps,
   type,
 }: TNavItem) => {
+
+  const handleLinkClick = useCallback(() => {
+    busDispatch<IEventBusClickLink>({ type: '@@-click-link', data: 'Hello World!' });
```
Comment (pivanov) [inline src/components/navigation/navigationItem.tsx:29]: to generic as I mention above

File: src/hooks/useResponsive.ts
Comment (pivanov) [inline src/hooks/useResponsive.ts:1]: first ... we have `utils/hooks` I've missed that we use `./src/hooks` we should use utils/hooks

File: src/hooks/useResponsive.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    isDesktop,
+    setIsDesktop,
+  ] = useState<boolean>(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      const width = window.innerWidth;
+      setIsMobile(width < breakpoints.mobile);
+      setIsTablet(width >= breakpoints.mobile && width < breakpoints.tablet);
+      setIsDesktop(width >= breakpoints.tablet);
```
Comment (pivanov) [inline src/hooks/useResponsive.ts:30]: use one setState we don't need to have 3 or more later

**Review by pivanov — CHANGES_REQUESTED** [review]
let's fix these first and we will check it latter deeply

File: src/components/navigation/navigationDropdown.tsx (hunk 4 above)
Comment (Denis-Dinkov) [reply src/components/navigation/navigationDropdown.tsx:45]: Done

**Review by pivanov — APPROVED** [review]
r+

**Pull request merged**